drivers/timer/intel_adsp_timer.c

```diff
@@ -217,13 +223,22 @@ void smp_timer_init(void)
 static int sys_clock_driver_init(void)
 {
 	uint64_t curr = count();
-
-	IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
+	if (!sys_clock_driver_reinit) {
+		IRQ_CONNECT(TIMER_IRQ, 0, compare_isr, 0, 0);
+	}
```

Comment:
This seems odd, what is it trying to fix?  IRQ_CONNECT() isn't code[1], it's a build-time construct that associates an ISR with an interrupt by emitting a record to a special linker section.

This is a noop, basically.  Are you maybe wanting to control irq_en/disable() instead?

[1] On most architectures, including xtensa.  There are a few that require something happen at runtime, but I'd have to look up the details.

Reply:
Sorry, the documentation of `#define IRQ_CONNECT`:

> Although this routine is invoked at run-time, all of its arguments must be
>  * computable by the compiler at build time.
>  * (...)

Made me think, that this function is called runtime.

Reply:
Yes, this is build-time construct as you said.
Left just the `sys_clock_idle_exit()` function that simply calls `sys_clock_driver_init()`.